File: go/set_trunk_alias.go
package main

import (
    "fmt"
    "os"
)

func main() {
    trunkSid := os.Getenv("TRUNK_SID")
    exophone := os.Getenv("EXOPHONE")
    
    if trunkSid == "" {
        fmt.Println("Error: TRUNK_SID is required. Set it in your .env file after creating a trunk.")
        os.Exit(1)
    }
    
    if exophone == "" {
        fmt.Println("Warning: EXOPHONE is not set. Skipping trunk alias configuration.")
        os.Exit(0)
    }
    
    fmt.Printf("Setting trunk alias %s for trunk %s...\n", exophone, trunkSid)
    
    payload := map[string]interface{}{
        "settings": []map[string]string{
            {"name": "trunk_external_alias", "value": exophone},
        },
    }
    
    post("/trunks/"+trunkSid+"/settings", payload)
    fmt.Println("Trunk alias set successfully!")
} 

File: java/SetTrunkAlias.java
public class SetTrunkAlias {
    public static void main(String[] args) throws Exception {
        String trunkSid = System.getenv("TRUNK_SID");
        String exophone = System.getenv("EXOPHONE");
        
        if (trunkSid == null || trunkSid.isEmpty()) {
            System.err.println("Error: TRUNK_SID is required. Set it in your .env file after creating a trunk.");
            System.exit(1);
        }
        
        if (exophone == null || exophone.isEmpty()) {
            System.out.println("Warning: EXOPHONE is not set. Skipping trunk alias configuration.");
            System.exit(0);
        }
        
        System.out.printf("Setting trunk alias %s for trunk %s...%n", exophone, trunkSid);
        
        String json = String.format("{\"settings\":[{\"name\":\"trunk_external_alias\",\"value\":\"%s\"}]}", exophone);
        _Client.post("/trunks/" + trunkSid + "/settings", json);
        
        System.out.println("Trunk alias set successfully!");
    }
} 